engine/src/main/java/org/hibernate/search/bridge/builtin/time/impl/LocalDateNumericBridge.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.bridge.builtin.time.impl;
+
+import java.time.LocalDate;
+
+/**
+ * Converts a {@link LocalDate} to a {@link Long}.
+ * <p>
+ * The digits of the resulting number will have the following pattern: yyyyMMdd
+ * <p>
+ * For example, the date 2001-3-25 will become the number 20010325.
```

Comment:
My concern is that this pattern may trick people into doing wrong
assumptions and date calculations based on the indexed values, e.g.
20120901-20120830 to determine the number of days between two dates.
Admittedly it's unlikely that someone accesses the raw indexed value, but
still it seems problematic to me. It's a bit like storing numbers to the
power of 8 in a numeric field to the power of 10.